src/pages/admin/AdminSessionPage.tsx

```diff
@@ -0,0 +1,217 @@
+    } catch (err) {
+      setError('Failed to fetch sessions');
+      console.error(err);
+      setSessions([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+    setError('');
+
+    if (!expiresDate) {
+      setError('Please select an expiration date');
+      return;
+    }
+
+    const expiresISO = expiresDate.toISOString();
+
+    try {
+      await api.post('/engagement/attendance/session', {
+        title,
+        key,
+        expires: expiresISO,
+      });
+
+      fetchSessions();
+      setTitle('');
+      setKey('');
+      setExpiresDate(null);
+    } catch (err: any) {
+      setError(err.response?.data?.error || 'Failed to create session');
+      console.error(err);
+    }
+  };
+
+  return (
+    <Box p={4}>
+      <VStack spacing={6} align="stretch">
+        <Button
+          as={Link}
+          to="/admin"
+          colorScheme="blue"
+          leftIcon={<ArrowBackIcon />}
+          w="fit-content"
+          mb="16px"
+        >
+          Go Back
+        </Button>
+
+        <Heading size="lg">Attendance Sessions</Heading>
+
+        <Card>
+          <CardHeader>
+            <HStack>
+              <Calendar size={25} />
+              <Text>Create New Session</Text>
+            </HStack>
+          </CardHeader>
+          <CardBody>
+            <form onSubmit={handleSubmit}>
+              <VStack spacing={4} align="stretch">
+                <FormControl isRequired>
+                  <FormLabel>Title</FormLabel>
+                  <Input
+                    value={title}
+                    onChange={(e) => setTitle(e.target.value)}
+                    placeholder="Session title"
+                  />
+                </FormControl>
+
+                <FormControl isRequired>
+                  <FormLabel>Key</FormLabel>
+                  <Input
+                    value={key}
+                    onChange={(e) => setKey(e.target.value)}
+                    placeholder="Attendance key"
+                  />
+                </FormControl>
+
+                <FormControl isRequired>
+                  <FormLabel>Expires</FormLabel>
+                  <DatePicker
+                    selected={expiresDate}
+                    onChange={(date: Date | null) => setExpiresDate(date)}
+                    showTimeSelect
+                    timeFormat="HH:mm"
+                    timeIntervals={5}
+                    timeCaption="time"
+                    dateFormat="MMMM d, yyyy h:mm aa"
+                    minDate={new Date()}
+                    customInput={<Input placeholder="Select date and time" />}
+                    className="chakra-input css-1kp110w"
+                  />
+                </FormControl>
+
+                <Button type="submit" colorScheme="blue" isLoading={loading}>
+                  Create Session
+                </Button>
+              </VStack>
```

Comment:
I would extract the functionality and UI for creating a session into its own component. I think the `AdminSessionPage` component is doing too much.